src/components/TodoContainer.js

```diff
@@ -0,0 +1,216 @@
+        { id: addedTodo.id, title: addedTodo.fields.title },
+      ]);
+      //  update the local state directly
+
+      // Handle errors during adding a todo
+    } catch (error) {
+      console.log("Error fetching data:", error.message);
+      return null;
+    }
+  };
+
+  //Delete items method DELETE
+  const removeTodo = async (id) => {
+    const options = {
+      method: "DELETE",
+      headers: {
+        Authorization: `Bearer ${process.env.REACT_APP_AIRTABLE_API_TOKEN}`,
+        "Content-Type": "application/json",
+      },
+    };
+    try {
+      const response = await fetch(`${AirtableUrl}/${id}`, options);
+      if (!response.ok) {
+        throw new Error(`Error has occurred: ${response.status}`);
+      }
+
+      fetchData(); //deleting items from todo list
+
+      return await response.json();
+    } catch (error) {
+      console.log(error.message);
+      return null;
+    }
+  };
+
+  //Checkbox  Method PATCH
+  const updateData = useCallback(async (id, newValue) => {
+    const options = {
+      method: "PATCH",
+      headers: {
+        "Content-Type": "application/json",
+        Authorization: `Bearer ${process.env.REACT_APP_AIRTABLE_API_TOKEN}`,
+        // "Authorization": `Bearer ${process.env.REACT_APP_AIRTABLE_API_KEY}`
+      },
+      body: JSON.stringify({
+        fields: {
+          CheckBox: newValue.CheckBox,
+        },
+      }),
+    };
+
+    console.log(`Updating record with id: ${id}, newValue: ${newValue}`);
+    try {
+      const response = await fetch(`${AirtableUrl}/${id}`, options);
+      if (!response.ok) {
+        const message = `Error has ocurred: ${response.status}`;
+        throw new Error(message);
+      }
+      const data = await response.json();
+      // console.log(data); // Log the response data to the console
+      // console.log(`Updating record with id: ${id}, newValue:`, newValue);
+      return data;
+    } catch (error) {
+      console.log("Error updating data: ", error);
+    }
+  }, []);
+
+  // Sorting by JavaScript
+  //   The sorting order is determined by the sortOrder state variable.
+  //   After sorting, the sorted list is set as the new state for todoList.
+  //
+
+  const sortData = useCallback(() => {
+    setTodoList((prevTodoList) => {
+      const sorted = [...prevTodoList].sort((objectA, objectB) => {
+        const titleA = objectA.title ? objectA.title.toUpperCase() : "";
+        const titleB = objectB.title ? objectB.title.toUpperCase() : "";
+
+        if (titleA < titleB) {
+          return sortOrder === "asc" ? -1 : 1;
+        } else if (titleA > titleB) {
+          return sortOrder === "asc" ? 1 : -1;
+        } else {
+          return 0;
+        }
+      });
+
+      // Only return the sorted array if it's different from the current one otherwise got an infinite loop
+      if (JSON.stringify(sorted) === JSON.stringify(prevTodoList)) {
+        return prevTodoList;
+      } else {
+        return sorted;
+      }
```

Comment:
Requires change:
This is causing an infinite loop because of the useEffect dependencies below (it's tracking `todoList` which updates each time `sortData` is ran, which is causing an infinite loop). 

A simpler approach would be to separate your sortData functionality from the `todoList` state being updated. For example, when you toggle the sorting, you can update the `sortOrder`. When the sortOrder changes, the `todoList` is sorted as a side effect:
```js
// toggle the sort order from "asc" to "desc" and vice versa
const handleSortToggle = () => setSortOrder(prevSortOrder => prevSortOrder === "asc" ? "desc" : "asc"));

// listen for the "sortOrder" changing, and resort the "todoList"
useEffect(() => {
   setTodoList((prevTodoList) => {
      return Array.from(prevTodoList).sort((todoA, todoB) => {
        // when you use a "?." after a property, it's called 
        // "optional chaining" which attempts to access a
        // potentially undefined property of an Object or Array
        const titleA = todoA.title?.toUpperCase();
        const titleB = todoB.title?.toUpperCase();

        if (titleA === titleB) return 0;

        if (titleA < titleB) return sortOrder === "asc" ? -1 : 1;

        return sortOrder === "asc" ? 1 : -1;
    });
  });
}, [sortOrder]);